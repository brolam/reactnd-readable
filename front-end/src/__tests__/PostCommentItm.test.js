import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme';
import PostCommentItem from '../components/PostCommentItem'

test('render without error', () => {
  ReactDOM.render(
    <PostCommentItem
      comment={comment}
      onClickEditButton={e => { }}
      onClickDeleteButton={e => { }}
    />, document.createElement('div'));
})

test('on click Edit button', () => {
  let isClickedEditButton = false;
  const postCommentItem = mount(
    <PostCommentItem
      comment={comment}
      onClickEditButton={e => { isClickedEditButton = true }}
      onClickDeleteButton={e => { }}
    />);
  postCommentItem.find('button [className="edit-button"]').simulate('click')
  expect(isClickedEditButton).toBe(true);
})

test('on click Delete button', () => {
  let isClickedDeleteButton = false;
  const postCommentItem = mount(
    <PostCommentItem
      comment={comment}
      onClickDeleteButton={e => { isClickedDeleteButton = true }}
      onClickEditButton={e => { }}
    />);
  postCommentItem.find('button [className="delete-button"]').simulate('click')
  expect(isClickedDeleteButton).toBe(true);
})

test('last Snapshot', () => {
  const postCommentItem = renderer.create(
    <PostCommentItem
      comment={comment}
      onClickEditButton={e => { }}
      onClickDeleteButton={e => { }}
    />);
  let tree = postCommentItem.toJSON();
  expect(tree).toMatchSnapshot();
})


const comment = {
  id: '894tuq4ut84ut8v4t8wun89g',
  parentId: "8xf0y6ziyjabvozdd253nd",
  timestamp: 1468166872634,
  body: 'Hi there! I am a COMMENT.',
  author: 'thingtwo',
  voteScore: 6,
  deleted: false,
  parentDeleted: false
}
