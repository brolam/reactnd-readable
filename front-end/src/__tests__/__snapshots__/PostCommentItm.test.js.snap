// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`last Snapshot 1`] = `
<div
  className="post-comment"
>
  <div
    className="post-comment-author"
  >
    by 
    thingtwo
     
    a year ago
  </div>
  <div>
    Hi there! I am a COMMENT.
  </div>
  <div
    className="post-comment-footer"
  >
    <button
      className="edit-button"
      href="/"
      onClick={[Function]}
    >
      Edit
    </button>
    <button
      className="delete-button"
      onClick={[Function]}
    >
      Delete
    </button>
    <button
      className="liked"
      onClick={[Function]}
    >
      Liked 
      <span>
        6
      </span>
    </button>
    <button
      className="not-liked"
      onClick={[Function]}
    >
      Not like
    </button>
  </div>
</div>
`;
